src/js/port_handler.js

```diff
@@ -242,12 +242,14 @@ PortHandler.selectPort = function(ports) {
     for (let i = 0; i < ports.length; i++) {
         const portName = ports[i].displayName;
         if (portName) {
-            if (portName.includes('STM') || portName.includes('CP210')) {
-                const pathSelect = ports[i].path;
-                if (OS === 'Windows' || (OS !== 'Windows' && pathSelect.includes('tty'))) {
-                    this.portPickerElement.val(pathSelect);
-                    console.log(`Porthandler detected device ${portName} on port: ${pathSelect}`);
-                }
+            const pathSelect = ports[i].path;
+            const isWindows = (OS === 'Windows');
+            const isTty = pathSelect.includes('tty');
+            const driverRecognized = portName.includes('STM') || portName.includes('CP210');
+            const legacyDriverRecognized = portName.includes('usb');
+            if ((driverRecognized && (isWindows || isTty)) || (legacyDriverRecognized && isTty)) {
```

Comment:
This is very hard to understand now as it mixes different types of conditions in different groupings. How about

```
if (isWindows && driverRecognized || isTty && (driverRecognized || legacyDriverRecognized)) {
```